File: frontend/src/types/adb.ts
export interface DeviceInfo {
  serial: string
  state: 'device' | 'offline' | 'unauthorized'
  product: string | null
  model: string | null
  transport_id: string | null
}

export interface IosDeviceInfo {
  udid: string
  name: string | null
  model: string | null
  ios_version: string | null
  jailbroken: boolean
}

export interface DynamicSession {
  id: number
  analysis_id: number
  created_at: string
  device_serial: string
  package_name: string
  status: 'active' | 'stopped'
  proxy_port: number | null
  frida_attached: boolean
}

export interface LogcatLine {
  ts: string
  level: 'V' | 'D' | 'I' | 'W' | 'E' | 'F'
  tag: string
  message: string
  pid?: string
}

File: backend/schemas/adb.py
from pydantic import BaseModel


class DeviceInfo(BaseModel):
    serial: str
    state: str  # device | offline | unauthorized
    product: str | None
    model: str | None
    transport_id: str | None


class InstallResult(BaseModel):
    success: bool
    message: str


class LaunchRequest(BaseModel):
    package_name: str
    activity: str | None = None


class LogcatLine(BaseModel):
    ts: str
    level: str  # V D I W E F
    tag: str
    message: str
    pid: str | None = None
